pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -329,6 +329,24 @@ static Py_ssize_t get_attr_length(PyObject *obj, char *attr) {
     return ret;
 }
 
+npy_int64 get_long_attr(PyObject *o, const char *attr) {
+  npy_int64 long_val;
+  PyObject *value = PyObject_GetAttrString(o, attr);
+  long_val = (PyLong_Check(value) ?
+              PyLong_AsLongLong(value) : PyInt_AS_LONG(value));
+  Py_DECREF(value);
+  return long_val;
+}
+
+npy_float64 total_seconds(PyObject *td) {
```

Comment:
There is no public c api for getting total seconds from a timedelta object (only for days, seconds, microseconds -> https://docs.python.org/3.3/c-api/datetime.html#PyDateTime_DELTA_GET_DAYS, which also is only new in python 3). So through the c-api this is not possible.

But, I suppose we can just get the python attribute `total_seconds` without using the c-api ? (or is that only possible in cython ?). 
@jbrockmendel does `npy_int64 days = get_long_attr(td, "total_seconds");` work?

Otherwise, I would just merge this as it is a net improvement